### PR TITLE
Dependency updates

### PR DESCRIPTION
**Pull request by mnonnenmacher**
See the commit messages for details.

### CODE REVIEW

File: plugin/src/functionalTest/groovy/com/here/gradle/plugins/jobdsl/JobDslPluginTest.groovy [hunk 1]
```diff
@@ -58,7 +58,7 @@ class JobDslPluginTest extends Specification {
 
         then:
         project.repositories.find { it.url.toString() == 'https://repo.jenkins-ci.org/releases/' }
-        project.repositories.find { it.url.toString() == 'https://repo1.maven.org/maven2/' }
+        project.repositories.find { it.url.toString() == 'https://repo.maven.apache.org/maven2/' }
```
Comment (sschuberth) [inline plugin/src/functionalTest/groovy/com/here/gradle/plugins/jobdsl/JobDslPluginTest.groovy]: Could you please explain in the commit message why this change is also necessary when upgrading Gradle?

File: plugin/build.gradle [hunk 2]
```diff
@@ -101,7 +101,7 @@ dependencies {
     // Plugins to be available for single tests using the @WithPlugin annotation.
     jenkinsPlugins 'org.jenkins-ci.plugins:cloudbees-folder:6.1.0'
     jenkinsPlugins 'org.jenkins-ci.plugins:gradle:1.22' // Old version used to test reporting of outdated plugins.
-    jenkinsPlugins 'org.jenkins-ci.plugins:htmlpublisher:1.12' // Used to test reporting of deprecated plugins.
+    jenkinsPlugins 'org.jenkins-ci.plugins:gradle:1.26' // Old version used to test reporting of deprecated plugins.
```
Comment (sschuberth) [inline plugin/build.gradle]: Is it worth mentioning in the commit message why this was also changed from `htmlpublisher`?

File: plugin/src/functionalTest/resources/updateJenkins/folder.xml [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
+        </views>
+        <tabBar class="hudson.views.DefaultViewsTabBar"/>
+        <primaryView>All</primaryView>
+    </folderViews>
     <healthMetrics>
-        <com.cloudbees.hudson.plugins.folder.health.WorstChildHealthMetric></com.cloudbees.hudson.plugins.folder.health.WorstChildHealthMetric>
+        <com.cloudbees.hudson.plugins.folder.health.WorstChildHealthMetric/>
     </healthMetrics>
-</com.cloudbees.hudson.plugins.folder.Folder>
+</com.cloudbees.hudson.plugins.folder.Folder>
```
Comment (sschuberth) [inline plugin/src/functionalTest/resources/updateJenkins/folder.xml]: Nit: No newline at end of file.

File: plugin/src/functionalTest/groovy/com/here/gradle/plugins/jobdsl/JobDslPluginTest.groovy (hunk 1 above)
Comment (mnonnenmacher) [reply plugin/src/functionalTest/groovy/com/here/gradle/plugins/jobdsl/JobDslPluginTest.groovy]: Done.

File: plugin/src/functionalTest/resources/updateJenkins/folder.xml (hunk 3 above)
Comment (mnonnenmacher) [reply plugin/src/functionalTest/resources/updateJenkins/folder.xml]: Done.

File: plugin/build.gradle (hunk 2 above)
Comment (sschuberth) [reply plugin/build.gradle]: @mnonnenmacher did you forget about this one?

Comment (mnonnenmacher) [reply plugin/build.gradle]: Actually I missed it. Done.
